src/movies.js

```diff
@@ -1,16 +1,133 @@
+// map, reduce, filter and sort
+
+
 // Iteration 1: All directors? - Get the array of all directors.
+function getAllDirectors(movies) {
+    return movies.map(function(movie) {
+        return movie["director"];
+    })
+}
+
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors. How could you "clean" a bit this array and make it unified (without duplicates)?
 
+function getAllUniqueDirectors(movies) {
+    return [...new Set(getAllDirectors(movies))];
+}
+
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
 
+function howManyMovies(movies) {
+    let spielbergMovies = movies.filter(function(movie) {
+        if (movie['director'] === 'Steven Spielberg' && movie["genre"].includes('Drama')) return true
```

Comment:
Instead of returning true, the condition can be returned.

Reply:
Would I just write it like `return if ...`?

Reply:
You can simply write:
`return movie['director'] === 'Steven Spielberg' && movie["genre"].includes('Drama');`